<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
  <meta http-equiv="Content-Style-Type" content="text/css">
  <title>readme</title>
  <meta name="Generator" content="Cocoa HTML Writer">
  <meta name="CocoaVersion" content="2022.2">
  <style type="text/css">
    p.p3 {margin: 0.0px 0.0px 16.0px 0.0px; line-height: 25.0px; font: 16.0px 'Helvetica Neue'; color: #262626; -webkit-text-stroke: #262626}
    p.p4 {margin: 0.0px 0.0px 16.0px 0.0px; line-height: 25.0px; font: 16.0px 'Helvetica Neue'; color: #262626; -webkit-text-stroke: #262626; min-height: 18.0px}
    p.p9 {margin: 0.0px 0.0px 0.0px 0.0px; line-height: 19.0px; font: 13.6px Consolas; color: #262626; -webkit-text-stroke: #262626; background-color: #f5f5f5}
    p.p10 {margin: 0.0px 0.0px 0.0px 0.0px; line-height: 19.0px; font: 13.6px Consolas; color: #262626; -webkit-text-stroke: #262626; background-color: #f5f5f5; min-height: 15.0px}
    li.li1 {margin: 0.0px 0.0px 0.0px 0.0px; line-height: 25.0px; font: 16.0px 'Helvetica Neue'; color: #3262b2; -webkit-text-stroke: #3262b2}
    li.li6 {margin: 0.0px 0.0px 0.0px 0.0px; line-height: 25.0px; font: 16.0px 'Helvetica Neue'; color: #262626; -webkit-text-stroke: #262626}
    li.li7 {margin: 0.0px 0.0px 0.0px 0.0px; line-height: 21.0px; font: 16.0px 'Helvetica Neue'; color: #262626; -webkit-text-stroke: #262626}
    li.li8 {margin: 0.0px 0.0px 0.0px 0.0px; line-height: 21.0px; font: 13.6px Consolas; color: #262626; -webkit-text-stroke: #262626}
    li.li11 {margin: 0.0px 0.0px 16.0px 0.0px; line-height: 21.0px; font: 13.6px Consolas; color: #262626; -webkit-text-stroke: #262626}
    li.li12 {margin: 0.0px 0.0px 16.0px 0.0px; line-height: 25.0px; font: 13.6px Consolas; color: #262626; -webkit-text-stroke: #262626}
    span.s1 {-webkit-text-stroke: 0px #000000}
    span.s2 {font-kerning: none}
    span.s3 {font: 13.6px Consolas; font-kerning: none; background-color: rgba(0, 0, 0, 0.039)}
    span.s4 {font: 13.6px Consolas; background-color: rgba(0, 0, 0, 0.039); -webkit-text-stroke: 0px #000000}
    span.s5 {background-color: rgba(0, 0, 0, 0.039); -webkit-text-stroke: 0px #000000}
    span.s6 {font-kerning: none; background-color: rgba(0, 0, 0, 0.039)}
    span.s7 {font: 16.0px 'Helvetica Neue'; font-kerning: none}
    span.s8 {font: 16.0px 'Helvetica Neue'; -webkit-text-stroke: 0px #000000}
    ol.ol1 {list-style-type: decimal}
    ul.ul1 {list-style-type: disc}
    ul.ul2 {list-style-type: circle}
  </style>
</head>
<body>
<ul class="ul1">
  <li class="li1"><span class="s1"></span><span class="s2">Integration of proteomics with genomics and transcriptomics increases the diagnosis rate of Mendelian disorders</span></li>
  <ul class="ul2">
    <li class="li1"><span class="s1"></span><span class="s2">Project structure</span></li>
    <li class="li1"><span class="s1"></span><span class="s2">Data and prerequisites</span></li>
    <li class="li1"><span class="s1"></span><span class="s2">Repository setup</span></li>
    <li class="li1"><span class="s1"></span><span class="s2">Run the full pipeline</span></li>
  </ul>
</ul>
<h1 style="margin: 0.0px 0.0px 16.0px 0.0px; line-height: 33.0px; font: 28.0px 'Helvetica Neue'; color: #262626; -webkit-text-stroke: #262626"><span class="s2"><b>Integration of proteomics with genomics and transcriptomics increases the diagnosis rate of Mendelian disorders</b></span></h1>
<p class="p3"><span class="s2">This project contains different scripts to automatize and visualize analysis performed for the “Integration of proteomics with genomics and transcriptomics increases the diagnosis rate of Mendelian disorders” <a href="https://drive.google.com/file/d/1y_AXrBBlkk0nG5TFiq4ayRu6qy2I3gLo/view?usp=sharing">paper</a>.</span></p>
<p class="p4"><span class="s2"></span><br></p>
<h2 style="margin: 0.0px 0.0px 16.0px 0.0px; line-height: 29.0px; font: 24.0px 'Helvetica Neue'; color: #262626; -webkit-text-stroke: #262626"><span class="s2"><b>Project structure</b></span></h2>
<p class="p3"><span class="s2">This project is setup as a <a href="https://github.com/gagneurlab/wBuild">wBuild workflow</a>. This is an automatic build tool for R reports based on <a href="https://snakemake.readthedocs.io/en/stable/">snakemake</a>.</span></p>
<ul class="ul1">
  <li class="li6"><span class="s1"></span><span class="s2">The </span><span class="s3">wbuild.yaml</span><span class="s2"> is the main configuration file to setup up the workflow</span></li>
  <li class="li6"><span class="s1"></span><span class="s2">The </span><span class="s3">Scripts</span><span class="s2"> folder contains scripts which will be rendered as HTML reports</span></li>
  <li class="li6"><span class="s1"></span><span class="s2">The </span><span class="s3">src</span><span class="s2"> folder contains additional helper functions and scripts</span></li>
  <li class="li6"><span class="s1"></span><span class="s2">The </span><span class="s3">Output</span><span class="s2"> folder will contain all files produced in the analysis pipeline</span></li>
  <ul class="ul2">
    <li class="li7"><span class="s4"></span><span class="s3">Output/html</span><span class="s2"> contains the final HTML report</span></li>
  </ul>
</ul>
<h2 style="margin: 0.0px 0.0px 16.0px 0.0px; line-height: 29.0px; font: 24.0px 'Helvetica Neue'; color: #262626; -webkit-text-stroke: #262626"><span class="s2"><b>Data and prerequisites</b></span></h2>
<p class="p3"><span class="s2">This project depends on the python packages </span><span class="s3">wBuild</span><span class="s2"> and </span><span class="s3">OUTRIDER2</span></p>
<p class="p3"><span class="s2">The pipeline starts with the series of files available via Zenodo: * </span><span class="s3">raw_data</span><span class="s2"> * </span><span class="s3">proteomics_annotation.tsv</span><span class="s2"> - sample annotation * </span><span class="s3">proteomics_not_normalized.tsv</span><span class="s2"> - Proteomics intensity matrix * </span><span class="s3">raw_counts.tsv</span><span class="s2"> - RNA-seq count matrix * </span><span class="s3">Patient_HPO_phenotypes.tsv</span><span class="s2"> - Phenotype data recorded using HPO terms for diagnosed cases. * </span><span class="s3">patient_variant_hpo_data.tsv</span><span class="s2"> - Gene annotation for all individuals. Since the genetic data are not publicly shareable, we provide only gene-level information.</span></p>
<ul class="ul1">
  <li class="li8"><span class="s5"></span><span class="s6">datasets</span></li>
  <ul class="ul2">
    <li class="li7"><span class="s4"></span><span class="s3">disease_genes.tsv</span><span class="s2"> - List of Mendelian disease genes aggregated from several studies. Downloaded automatically:</span></li>
    <li class="li7"><span class="s4"></span><span class="s3">gencode.v29lift37.annotation.gtf.gz</span><span class="s2"> - Gene-level model based on the <a href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/GRCh37_mapping/gencode.v29lift37.annotation.gtf.gz">GENCODE 29 transcript model</a></span></li>
    <li class="li7"><span class="s4"></span><span class="s3">Table_S1_gene_info_at_protein_level.xlsx</span><span class="s2"> - Supplementary Tble1 from <a href="https://www.cell.com/cell/fulltext/S0092-8674(20)31078-3?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867420310783%3Fshowall%3Dtrue">GTEx proteomics study Jiang et al, 2020, Cell</a> Data is available at the <a href="https://storage.googleapis.com/gtex_egtex/proteomics/Table_S1_gene_info_at_protein_level.xlsx">GTEx page</a></span></li>
    <li class="li7"><span class="s4"></span><span class="s3">allComplexes.txt</span><span class="s2"> - CORUM protein complexes, available at CORUM <a href="http://mips.helmholtz-muenchen.de/corum/download/allComplexes.txt.zip">web page</a></span></li>
  </ul>
</ul>
<h2 style="margin: 0.0px 0.0px 16.0px 0.0px; line-height: 29.0px; font: 24.0px 'Helvetica Neue'; color: #262626; -webkit-text-stroke: #262626"><span class="s2"><b>Repository setup</b></span></h2>
<p class="p3"><span class="s2">First download the repo and its dependencies:</span></p>
<p class="p9"><span class="s2"># outrider2 package for python omplementation of PRTRIDER</span></p>
<p class="p9"><span class="s2">git clone --branch outrider2 git@gitlab.cmm.in.tum.de:brechtma/OUTRIDER.git</span></p>
<p class="p9"><span class="s2">git clone --branch outrider2 git@gitlab.cmm.in.tum.de:gagneurlab/py_outrider.git</span></p>
<p class="p9"><span class="s2">ln -s OUTRIDER/inst/python py_outrider/py_outrider</span></p>
<p class="p9"><span class="s2">ln -s /data/nasif12/home_if12/smirnovd/omicsDiagnostics/OUTRIDER/inst/python /data/nasif12/home_if12/smirnovd/omicsDiagnostics/py_outrider/py_outrider</span></p>
<p class="p10"><span class="s2"></span><br></p>
<p class="p9"><span class="s2"># analysis code</span></p>
<p class="p9"><span class="s2">git clone https://github.com/gagneurlab/omicsDiagnostics</span></p>
<p class="p9"><span class="s2">cd omicsDiagnostics</span></p>
<p class="p3"><span class="s2">and install wbuild using pip by running.</span></p>
<p class="p9"><span class="s2">pip install wBuild</span></p>
<p class="p9"><span class="s2">wBuild init</span></p>
<p class="p3"><span class="s2">Since </span><span class="s3">wBuild init</span><span class="s2"> will reset the current </span><span class="s3">Snakefile</span><span class="s2">, </span><span class="s3">readme.md</span><span class="s2">, and </span><span class="s3">wbuild.yaml</span><span class="s2"> we have to revert them again with git.</span></p>
<p class="p9"><span class="s2">git checkout Snakefile</span></p>
<p class="p9"><span class="s2">git checkout wbuild.yaml</span></p>
<p class="p9"><span class="s2">git checkout readme.md</span></p>
<h2 style="margin: 0.0px 0.0px 16.0px 0.0px; line-height: 29.0px; font: 24.0px 'Helvetica Neue'; color: #262626; -webkit-text-stroke: #262626"><span class="s2"><b>Run the full pipeline</b></span></h2>
<p class="p3"><span class="s2">To run the full pipeline, execute the following commands with 10 cores in parallel:</span></p>
<ol class="ol1">
  <li class="li11"><span class="s5"></span><span class="s6">conda env create --name omicsDiagnostics --file=enviorment.yml</span><span class="s7"><br>
</span></li>
  <li class="li11"><span class="s5"></span><span class="s6">conda activate omicsDiagnostics</span><span class="s7"><br>
</span></li>
  <li class="li12"><span class="s8"></span><span class="s7">Install R::gganatogram package with </span><span class="s6">devtools::install_github("jespermaag/gganatogram")</span><span class="s7"><br>
</span></li>
  <li class="li11"><span class="s5"></span><span class="s6">snakemake publish -j 10</span><span class="s7"><br>
</span></li>
</ol>
</body>
</html>
