<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="smirnovd" />

<meta name="date" content="2021-01-27" />

<title>Patient analysis</title>

<script src="libR/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libR/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libR/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libR/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="libR/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libR/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libR/navigation-1.1/tabsets.js"></script>
<script src="libR/navigation-1.1/codefolding.js"></script>
<script src="libR/navigation-1.1/sourceembed.js"></script>
<link href="libR/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libR/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<link rel="stylesheet" href="lib/add_content_table.css" type="text/css" />
<link rel="stylesheet" href="lib/leo_style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Patient analysis</h1>
<h4 class="author">smirnovd</h4>
<h4 class="date">2021-01-27</h4>

</div>


<pre class="r"><code># Load config
source(snakemake@input$config)

# Load plotting functions
source(&quot;src/functions/plots.R&quot;)



# Load sample annotation
# sa &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/proteomics_annotation.tsv&#39;)
sa &lt;- fread(snakemake@input$sample_annotation)
sa &lt;- sa[USE_FOR_PROTEOMICS_PAPER == T]
# Subset diagnosed cases and candidates
sa &lt;- sa[ !is.na(KNOWN_MUTATION)]


# Read integrated omics file 
# rp &lt;- readRDS(&quot;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/integration/patient_omics.RDS&quot;) %&gt;% as.data.table()
rp &lt;- readRDS(snakemake@input$patient_omics) %&gt;% as.data.table()
# Subset diagnosed cases and candidates
rp &lt;- rp[SAMPLE_ID %in% sa$SAMPLE_ID]

# Subset only outliers or causal/candidate genes
rp &lt;- rp[ outlier_class != &#39;non_outlier&#39; | causal_gene == T]


# Load patient&#39;s HPO 
pat_hpo &lt;- fread(snakemake@input$patient_hpo)
# pat_hpo &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/Patient_HPO_phenotypes.tsv&#39;)

# Subset diagnosed cases and candidates
pat_hpo &lt;- pat_hpo[SAMPLE_ID %in% sa$SAMPLE_ID, c(&quot;SAMPLE_ID&quot;, &quot;HPO_term&quot;)]
pat_hpo &lt;-pat_hpo[!duplicated(pat_hpo ),]

# Load patient&#39;s affected organ systems
patients &lt;- fread(snakemake@input$patient_organ_syst)
# patients &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/HPO/Patients_affected_organs.tsv&#39;)
# Subset diagnosed cases and candidates
patients &lt;- patients[SAMPLE_ID %in% sa$SAMPLE_ID]</code></pre>
<div id="category-i---known-pathogenic-variants" class="section level2">
<h2>Category I - Known pathogenic variants</h2>
<pre class="r"><code>for (i in unique(sa[ CATEGORY == &quot;I&quot; ]$SAMPLE_ID)){
  print(paste(&quot;Sample ID:&quot;, i, &quot;Causal gene:&quot;, sa[SAMPLE_ID == i]$KNOWN_MUTATION ) )
  
  plot_org &lt;- gganatogram(data=  patients[SAMPLE_ID == i], 
                          fillOutline=&#39;white&#39;, 
                          organism=&#39;human&#39;, 
                          sex= unique(sa[SAMPLE_ID == i]$gender), 
                          fill=&quot;colour&quot;)+ 
    theme_void() + ggtitle(paste0(&quot;individual #&quot;, i )) +
    theme(plot.title = element_text(margin = NULL,face=&quot;bold&quot;, size = 9, hjust = 0.5),
          plot.margin = margin(0, 0, 0, 0, &quot;cm&quot;))
  
  phenotypes &lt;- pat_hpo[SAMPLE_ID == i,  c( &quot;HPO_term&quot;) ]
  phenotypes &lt;- gridExtra::tableGrob(phenotypes)
  plot_org &lt;-  plot_org  | phenotypes 
  
  if (nrow(rp[SAMPLE_ID == i ]) == 0){
    print(&quot;No outliers detected&quot;)
    plot_sample &lt;- plot_org

  }else{
    plot_sample &lt;- plot_patient_all(rp, sample = i) 
    plot_sample &lt;- plot_org | plot_sample
  }
  print(plot_sample)
}</code></pre>
<pre><code>## [1] &quot;Sample ID: OM46241 Causal gene: SSBP1&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-1.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM73953 Causal gene: GNAO1&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-2.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM42562 Causal gene: ADPRHL2&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-3.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM71746 Causal gene: MDH2&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-4.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM37109 Causal gene: YME1L1&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-5.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM51049 Causal gene: UQCRFS1&quot;
## [1] &quot;No outliers detected&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-6.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM48652 Causal gene: NBAS&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-7.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM88146 Causal gene: NBAS&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-8.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM73546 Causal gene: TIMMDC1&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-9.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM75219_1 Causal gene: ELAC2&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-10.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM33917 Causal gene: FOXRED1&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-11.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM01282 Causal gene: TSFM&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-12.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM65728 Causal gene: ACAD9&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-13.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM88728 Causal gene: STIM1&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-14.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM57274 Causal gene: TRMU&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-15.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM20680 Causal gene: BCS1L&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-2-16.png" width="1728" /></p>
</div>
<div id="category-ii---vus" class="section level2">
<h2>Category II - VUS</h2>
<div id="category-iia---vus-validated" class="section level3">
<h3>Category IIa - VUS validated</h3>
<pre class="r"><code>for (i in unique(sa[ CATEGORY == &quot;IIa&quot; ]$SAMPLE_ID)){
  print(paste(&quot;Sample ID:&quot;, i, &quot;Causal gene:&quot;, sa[SAMPLE_ID == i]$KNOWN_MUTATION ) )
  
  plot_org &lt;- gganatogram(data=  patients[SAMPLE_ID == i], 
                          fillOutline=&#39;white&#39;, 
                          organism=&#39;human&#39;, 
                          sex= unique(sa[SAMPLE_ID == i]$gender), 
                          fill=&quot;colour&quot;)+ 
    theme_void() + ggtitle(paste0(&quot;individual #&quot;, i )) +
    theme(plot.title = element_text(margin = NULL,face=&quot;bold&quot;, size = 9, hjust = 0.5),
          plot.margin = margin(0, 0, 0, 0, &quot;cm&quot;))
  
  phenotypes &lt;- pat_hpo[SAMPLE_ID == i,  c( &quot;HPO_term&quot;) ]
  phenotypes &lt;- gridExtra::tableGrob(phenotypes)
  plot_org &lt;-  plot_org  | phenotypes 
  
  if (nrow(rp[SAMPLE_ID == i ]) == 0){
    print(&quot;No outliers detected&quot;)
    plot_sample &lt;- plot_org
  }else{
    plot_sample &lt;- plot_patient_all(rp, sample = i) 
    plot_sample &lt;- plot_org | plot_sample
  }
  print(plot_sample)
}</code></pre>
<pre><code>## [1] &quot;Sample ID: OM23417 Causal gene: FDXR&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-1.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM65708 Causal gene: ACAD9&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-2.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM43933 Causal gene: DNAJC30&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-3.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM02364 Causal gene: SLC25A4&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-4.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM72697 Causal gene: DNAJC30&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-5.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM70698 Causal gene: TXNIP&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-6.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM50946 Causal gene: NDUFB11&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-7.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM27175 Causal gene: FH&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-8.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM26649 Causal gene: AARS2&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-9.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM77629 Causal gene: ETFDH&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-10.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM61355 Causal gene: YARS&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-11.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM05252 Causal gene: EARS2&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-12.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM88470 Causal gene: PC&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-13.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM20348 Causal gene: SUCLA2&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-3-14.png" width="1728" /></p>
</div>
<div id="category-iib---vus-not-validated" class="section level3">
<h3>Category IIb - VUS not validated</h3>
<pre class="r"><code>for (i in unique(sa[ CATEGORY == &quot;IIb&quot; ]$SAMPLE_ID)){
  print(paste(&quot;Sample ID:&quot;, i, &quot;Causal gene:&quot;, sa[SAMPLE_ID == i]$KNOWN_MUTATION ) )
  
  plot_org &lt;- gganatogram(data=  patients[SAMPLE_ID == i], 
                          fillOutline=&#39;white&#39;, 
                          organism=&#39;human&#39;, 
                          sex= unique(sa[SAMPLE_ID == i]$gender), 
                          fill=&quot;colour&quot;)+ 
    theme_void() + ggtitle(paste0(&quot;individual #&quot;, i )) +
    theme(plot.title = element_text(margin = NULL,face=&quot;bold&quot;, size = 9, hjust = 0.5),
          plot.margin = margin(0, 0, 0, 0, &quot;cm&quot;))
  
  phenotypes &lt;- pat_hpo[SAMPLE_ID == i,  c( &quot;HPO_term&quot;) ]
  phenotypes &lt;- gridExtra::tableGrob(phenotypes)
  plot_org &lt;-  plot_org  | phenotypes 
  
  if (nrow(rp[SAMPLE_ID == i ]) == 0){
    print(&quot;No outliers detected&quot;)
    plot_sample &lt;- plot_org
    
  }else{
    plot_sample &lt;- plot_patient_all(rp, sample = i) 
    plot_sample &lt;- plot_org | plot_sample
  }
  print(plot_sample)
}</code></pre>
<pre><code>## [1] &quot;Sample ID: OM94976 Causal gene: DNAJC30&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-4-1.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM00762 Causal gene: MRPL53&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-4-2.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM59244 Causal gene: TRMT5&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-4-3.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM25068 Causal gene: SLC25A4&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-4-4.png" width="1728" /></p>
</div>
<div id="category-iic---vus-not-detected" class="section level3">
<h3>Category IIc - VUS not detected</h3>
<pre class="r"><code>for (i in unique(sa[ CATEGORY == &quot;IIc&quot; ]$SAMPLE_ID)){
  print(paste(&quot;Sample ID:&quot;, i, &quot;Causal gene:&quot;, sa[SAMPLE_ID == i]$KNOWN_MUTATION ) )
  
  plot_org &lt;- gganatogram(data=  patients[SAMPLE_ID == i], 
                          fillOutline=&#39;white&#39;, 
                          organism=&#39;human&#39;, 
                          sex= unique(sa[SAMPLE_ID == i]$gender), 
                          fill=&quot;colour&quot;)+ 
    theme_void() + ggtitle(paste0(&quot;individual #&quot;, i )) +
    theme(plot.title = element_text(margin = NULL,face=&quot;bold&quot;, size = 9, hjust = 0.5),
          plot.margin = margin(0, 0, 0, 0, &quot;cm&quot;))
  
  phenotypes &lt;- pat_hpo[SAMPLE_ID == i,  c( &quot;HPO_term&quot;) ]
  phenotypes &lt;- gridExtra::tableGrob(phenotypes)
  plot_org &lt;-  plot_org  | phenotypes 
  
  if (nrow(rp[SAMPLE_ID == i ]) == 0){
    print(&quot;No outliers detected&quot;)
    plot_sample &lt;- plot_org
    
  }else{
    plot_sample &lt;- plot_patient_all(rp, sample = i) 
    plot_sample &lt;- plot_org | plot_sample
  }
  print(plot_sample)
}</code></pre>
<pre><code>## [1] &quot;Sample ID: OM66072 Causal gene: C19ORF70&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-5-1.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM30476 Causal gene: TASP1&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-5-2.png" width="1728" /></p>
</div>
</div>
<div id="category-iii-new-discoveries" class="section level2">
<h2>Category III New discoveries</h2>
<pre class="r"><code>for (i in unique(sa[ CATEGORY == &quot;III&quot; ]$SAMPLE_ID)){
  print(paste(&quot;Sample ID:&quot;, i, &quot;Causal gene:&quot;, sa[SAMPLE_ID == i]$KNOWN_MUTATION ) )
  
  plot_org &lt;- gganatogram(data=  patients[SAMPLE_ID == i], 
                          fillOutline=&#39;white&#39;, 
                          organism=&#39;human&#39;, 
                          sex= unique(sa[SAMPLE_ID == i]$gender), 
                          fill=&quot;colour&quot;)+ 
    theme_void() + ggtitle(paste0(&quot;individual #&quot;, i )) +
    theme(plot.title = element_text(margin = NULL,face=&quot;bold&quot;, size = 9, hjust = 0.5),
          plot.margin = margin(0, 0, 0, 0, &quot;cm&quot;))
  
  phenotypes &lt;- pat_hpo[SAMPLE_ID == i,  c( &quot;HPO_term&quot;) ]
  phenotypes &lt;- gridExtra::tableGrob(phenotypes)
  plot_org &lt;-  plot_org  | phenotypes 
  
  if (nrow(rp[SAMPLE_ID == i ]) == 0){
    print(&quot;No outliers detected&quot;)
    plot_sample &lt;- plot_org
    
  }else{
    plot_sample &lt;- plot_patient_all(rp, sample = i) 
    plot_sample &lt;- plot_org | plot_sample
  }
  print(plot_sample)
}</code></pre>
<pre><code>## [1] &quot;Sample ID: OM35261 Causal gene: MIB1&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-1.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM06865 Causal gene: EPG5&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-2.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM91786 Causal gene: LIG3&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-3.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM87369 Causal gene: NDUFB11&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-4.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM27390 Causal gene: MORC2&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-5.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM34814 Causal gene: RDH11&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-6.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM57837 Causal gene: MRPL38&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-7.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM38813 Causal gene: TIMMDC1&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-8.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM21111 Causal gene: NFU1&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-9.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM58230 Causal gene: VPS33B&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-10.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM75740 Causal gene: DARS2&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-11.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM03592 Causal gene: MRPL44&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-12.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM56706 Causal gene: GMPPB&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-13.png" width="1728" /></p>
<pre><code>## [1] &quot;Sample ID: OM36526 Causal gene: MRPS25&quot;</code></pre>
<p><img src="Scripts_Patient_analysis_Patient_analysis_files/figure-html/unnamed-chunk-6-14.png" width="1728" /></p>
</div>

<div id="rmd-source-code">#'---
#' title: Patient analysis 
#' author: smirnovd
#' wb:
#'  input:
#'  - config: 'src/config.R'
#'  - sample_annotation: '`sm config["ANNOTATION"]`'
#'  - patient_omics: '`sm config["PROC_DATA"] + "/integration/patient_omics.RDS"`'
#'  - patient_hpo: '`sm config["RAW_HPO"]`'
#'  - patient_organ_syst: '`sm config["PROC_DATA"] + "/HPO/Patients_affected_organs.tsv"`'
#'  output:
#' output: 
#'   html_document:
#'    code_folding: hide
#'    code_download: TRUE
#'---

# Load config
source(snakemake@input$config)

# Load plotting functions
source("src/functions/plots.R")



# Load sample annotation
# sa <- fread('/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/proteomics_annotation.tsv')
sa <- fread(snakemake@input$sample_annotation)
sa <- sa[USE_FOR_PROTEOMICS_PAPER == T]
# Subset diagnosed cases and candidates
sa <- sa[ !is.na(KNOWN_MUTATION)]


# Read integrated omics file 
# rp <- readRDS("/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/integration/patient_omics.RDS") %>% as.data.table()
rp <- readRDS(snakemake@input$patient_omics) %>% as.data.table()
# Subset diagnosed cases and candidates
rp <- rp[SAMPLE_ID %in% sa$SAMPLE_ID]

# Subset only outliers or causal/candidate genes
rp <- rp[ outlier_class != 'non_outlier' | causal_gene == T]


# Load patient's HPO 
pat_hpo <- fread(snakemake@input$patient_hpo)
# pat_hpo <- fread('/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/Patient_HPO_phenotypes.tsv')

# Subset diagnosed cases and candidates
pat_hpo <- pat_hpo[SAMPLE_ID %in% sa$SAMPLE_ID, c("SAMPLE_ID", "HPO_term")]
pat_hpo <-pat_hpo[!duplicated(pat_hpo ),]

# Load patient's affected organ systems
patients <- fread(snakemake@input$patient_organ_syst)
# patients <- fread('/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/HPO/Patients_affected_organs.tsv')
# Subset diagnosed cases and candidates
patients <- patients[SAMPLE_ID %in% sa$SAMPLE_ID]




#' ## Category I - Known pathogenic variants
#+ fig.width=18, fig.height=10
for (i in unique(sa[ CATEGORY == "I" ]$SAMPLE_ID)){
  print(paste("Sample ID:", i, "Causal gene:", sa[SAMPLE_ID == i]$KNOWN_MUTATION ) )
  
  plot_org <- gganatogram(data=  patients[SAMPLE_ID == i], 
                          fillOutline='white', 
                          organism='human', 
                          sex= unique(sa[SAMPLE_ID == i]$gender), 
                          fill="colour")+ 
    theme_void() + ggtitle(paste0("individual #", i )) +
    theme(plot.title = element_text(margin = NULL,face="bold", size = 9, hjust = 0.5),
          plot.margin = margin(0, 0, 0, 0, "cm"))
  
  phenotypes <- pat_hpo[SAMPLE_ID == i,  c( "HPO_term") ]
  phenotypes <- gridExtra::tableGrob(phenotypes)
  plot_org <-  plot_org  | phenotypes 
  
  if (nrow(rp[SAMPLE_ID == i ]) == 0){
    print("No outliers detected")
    plot_sample <- plot_org

  }else{
    plot_sample <- plot_patient_all(rp, sample = i) 
    plot_sample <- plot_org | plot_sample
  }
  print(plot_sample)
}




#' ## Category II - VUS

#' ### Category IIa - VUS validated
#+ fig.width=18, fig.height=10
for (i in unique(sa[ CATEGORY == "IIa" ]$SAMPLE_ID)){
  print(paste("Sample ID:", i, "Causal gene:", sa[SAMPLE_ID == i]$KNOWN_MUTATION ) )
  
  plot_org <- gganatogram(data=  patients[SAMPLE_ID == i], 
                          fillOutline='white', 
                          organism='human', 
                          sex= unique(sa[SAMPLE_ID == i]$gender), 
                          fill="colour")+ 
    theme_void() + ggtitle(paste0("individual #", i )) +
    theme(plot.title = element_text(margin = NULL,face="bold", size = 9, hjust = 0.5),
          plot.margin = margin(0, 0, 0, 0, "cm"))
  
  phenotypes <- pat_hpo[SAMPLE_ID == i,  c( "HPO_term") ]
  phenotypes <- gridExtra::tableGrob(phenotypes)
  plot_org <-  plot_org  | phenotypes 
  
  if (nrow(rp[SAMPLE_ID == i ]) == 0){
    print("No outliers detected")
    plot_sample <- plot_org
  }else{
    plot_sample <- plot_patient_all(rp, sample = i) 
    plot_sample <- plot_org | plot_sample
  }
  print(plot_sample)
}


#' ### Category IIb - VUS not validated
#+ fig.width=18, fig.height=10
for (i in unique(sa[ CATEGORY == "IIb" ]$SAMPLE_ID)){
  print(paste("Sample ID:", i, "Causal gene:", sa[SAMPLE_ID == i]$KNOWN_MUTATION ) )
  
  plot_org <- gganatogram(data=  patients[SAMPLE_ID == i], 
                          fillOutline='white', 
                          organism='human', 
                          sex= unique(sa[SAMPLE_ID == i]$gender), 
                          fill="colour")+ 
    theme_void() + ggtitle(paste0("individual #", i )) +
    theme(plot.title = element_text(margin = NULL,face="bold", size = 9, hjust = 0.5),
          plot.margin = margin(0, 0, 0, 0, "cm"))
  
  phenotypes <- pat_hpo[SAMPLE_ID == i,  c( "HPO_term") ]
  phenotypes <- gridExtra::tableGrob(phenotypes)
  plot_org <-  plot_org  | phenotypes 
  
  if (nrow(rp[SAMPLE_ID == i ]) == 0){
    print("No outliers detected")
    plot_sample <- plot_org
    
  }else{
    plot_sample <- plot_patient_all(rp, sample = i) 
    plot_sample <- plot_org | plot_sample
  }
  print(plot_sample)
}




#' ### Category IIc - VUS not detected
#+ fig.width=18, fig.height=10
for (i in unique(sa[ CATEGORY == "IIc" ]$SAMPLE_ID)){
  print(paste("Sample ID:", i, "Causal gene:", sa[SAMPLE_ID == i]$KNOWN_MUTATION ) )
  
  plot_org <- gganatogram(data=  patients[SAMPLE_ID == i], 
                          fillOutline='white', 
                          organism='human', 
                          sex= unique(sa[SAMPLE_ID == i]$gender), 
                          fill="colour")+ 
    theme_void() + ggtitle(paste0("individual #", i )) +
    theme(plot.title = element_text(margin = NULL,face="bold", size = 9, hjust = 0.5),
          plot.margin = margin(0, 0, 0, 0, "cm"))
  
  phenotypes <- pat_hpo[SAMPLE_ID == i,  c( "HPO_term") ]
  phenotypes <- gridExtra::tableGrob(phenotypes)
  plot_org <-  plot_org  | phenotypes 
  
  if (nrow(rp[SAMPLE_ID == i ]) == 0){
    print("No outliers detected")
    plot_sample <- plot_org
    
  }else{
    plot_sample <- plot_patient_all(rp, sample = i) 
    plot_sample <- plot_org | plot_sample
  }
  print(plot_sample)
}


#' ## Category III New discoveries
#+ fig.width=18, fig.height=10
for (i in unique(sa[ CATEGORY == "III" ]$SAMPLE_ID)){
  print(paste("Sample ID:", i, "Causal gene:", sa[SAMPLE_ID == i]$KNOWN_MUTATION ) )
  
  plot_org <- gganatogram(data=  patients[SAMPLE_ID == i], 
                          fillOutline='white', 
                          organism='human', 
                          sex= unique(sa[SAMPLE_ID == i]$gender), 
                          fill="colour")+ 
    theme_void() + ggtitle(paste0("individual #", i )) +
    theme(plot.title = element_text(margin = NULL,face="bold", size = 9, hjust = 0.5),
          plot.margin = margin(0, 0, 0, 0, "cm"))
  
  phenotypes <- pat_hpo[SAMPLE_ID == i,  c( "HPO_term") ]
  phenotypes <- gridExtra::tableGrob(phenotypes)
  plot_org <-  plot_org  | phenotypes 
  
  if (nrow(rp[SAMPLE_ID == i ]) == 0){
    print("No outliers detected")
    plot_sample <- plot_org
    
  }else{
    plot_sample <- plot_patient_all(rp, sample = i) 
    plot_sample <- plot_org | plot_sample
  }
  print(plot_sample)
}




</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("Patient_analysis.R");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
