<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="smirnovd" />

<meta name="date" content="2021-01-25" />

<title>Gene semantic similarity</title>

<script src="libR/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libR/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libR/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libR/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="libR/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libR/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libR/navigation-1.1/tabsets.js"></script>
<script src="libR/navigation-1.1/codefolding.js"></script>
<script src="libR/navigation-1.1/sourceembed.js"></script>
<link href="libR/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libR/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<link rel="stylesheet" href="lib/add_content_table.css" type="text/css" />
<link rel="stylesheet" href="lib/leo_style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Gene semantic similarity</h1>
<h4 class="author">smirnovd</h4>
<h4 class="date">2021-01-25</h4>

</div>


<pre class="r"><code>source(snakemake@input$config)
#source(&quot;src/config.R&quot;)


# READ ANNOTATION
# sa &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/proteomics_annotation.tsv&#39;)
sa &lt;- fread(snakemake@input$sa)
sa &lt;- sa[USE_FOR_PROTEOMICS_PAPER == T]


# load HPO ontology
# data(hpo) 
hpo &lt;- get_ontology(&quot;http://purl.obolibrary.org/obo/hp.obo&quot;, extract_tags=&quot;everything&quot;)


# Load patient&#39;s HPO 
pat_hpo &lt;- fread(snakemake@input$patient_hpo)
# pat_hpo &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/Patient_HPO_phenotypes.tsv&#39;)


pat_hpo &lt;- pat_hpo[, c(&quot;SAMPLE_ID&quot;, &quot;HPO_ID&quot;)]
# renmove duplicates from HPOs per patient 
pat_hpo &lt;-pat_hpo[!duplicated(pat_hpo ),]
pat_hpo &lt;-pat_hpo[HPO_ID %in%  unique(hpo$id)]


hpo_gene &lt;- fread(&quot;http://compbio.charite.de/jenkins/job/hpo.annotations/lastSuccessfulBuild/artifact/util/annotation/phenotype_to_genes.txt&quot;)
colnames(hpo_gene) &lt;- c(&quot;HPO_ID&quot;, &quot;HPO_name&quot;, &quot;entrezID&quot;, &quot;geneID&quot;, &quot;Additional_Info&quot;,  &quot;source&quot;, &quot;disease-ID&quot;)
hpo_gene &lt;- hpo_gene[, c(&quot;geneID&quot;, &quot;HPO_ID&quot;)]
hpo_gene &lt;- hpo_gene[!duplicated(hpo_gene)]
hpo_gene &lt;-hpo_gene[HPO_ID %in%  unique(hpo$id)]


# Fix gene names
hpo_gene[geneID == &quot;YARS1&quot;, geneID := &quot;YARS&quot; ]
hpo_gene[geneID == &quot;MICOS13&quot;, geneID := &quot;C19ORF70&quot; ]
hpo_gene[geneID == &quot;ADPRS&quot;, geneID := &quot;ADPRHL2&quot; ]


# Add known genotype - phenotype associations 
hpo_gene &lt;- rbind(hpo_gene, data.table( geneID = &quot;DNAJC30&quot;, HPO_ID = c(&quot;HP:0002076&quot;, &quot;HP:0000648&quot;, &quot;HP:0000505&quot;, &quot;HP:0011462&quot;, &quot;HP:0001112&quot;, &quot;HP:0003581&quot;, &quot;HP:0001263&quot;, &quot;HP:0001332&quot;, &quot;HP:0001251&quot;, &quot;HP:0002134&quot;, &quot;HP:0011923&quot;, &quot;HP:0011463&quot;  )  ) )
hpo_gene &lt;- rbind(hpo_gene, data.table( geneID = &quot;MRPS25&quot;, HPO_ID = c(&quot;HP:0003581&quot;, &quot;HP:0011445&quot;, &quot;HP:0001274&quot;, &quot;HP:0002421&quot;, &quot;HP:0001266&quot;, &quot;HP:0002509&quot;, &quot;HP:0001348&quot;, &quot;HP:0001274&quot;, &quot;HP:0007333&quot;, &quot;HP:0001324&quot;, 
                                                                      &quot;HP:0001385&quot;, &quot;HP:0000846&quot;, &quot;HP:0004322&quot;, &quot;HP:0000252&quot;, &quot;HP:0011924&quot;, &quot;HP:0008347&quot;, &quot;HP:0001270&quot;, &quot;HP:0003676&quot;, &quot;HP:0012378&quot;, &quot;HP:0002015&quot;)  ) ) 
hpo_gene &lt;- rbind(hpo_gene, data.table( geneID = &quot;TXNIP&quot;, HPO_ID = c(&quot;HP:0004902&quot;, &quot;HP:0001943&quot;, &quot;HP:0003355&quot;, &quot;HP:0006568&quot;, &quot;HP:0003658&quot;, &quot;HP:0004322&quot;, &quot;HP:0001508&quot;,&quot;HP:0001252&quot;, &quot;HP:0011968&quot;)  ) )
hpo_gene &lt;- rbind(hpo_gene, data.table( geneID = &quot;YARS&quot;, HPO_ID = c(&quot;HP:0001510&quot;, &quot;HP:0001263&quot;, &quot;HP:0007305&quot;, &quot;HP:0000407&quot;, &quot;HP:0000639&quot;, &quot;HP:0002611&quot;, &quot;HP:0001738&quot;, &quot;HP:0001943&quot;, &quot;HP:0001903&quot;, &quot;HP:0000093&quot;, &quot;HP:0100806&quot;, &quot;HP:0006528&quot;)  ) )

hpo_gene &lt;- hpo_gene[!duplicated(hpo_gene)]
sa &lt;- sa[SAMPLE_ID %in% unique(pat_hpo$SAMPLE_ID) ]

# extract ancestors
hp_ancestors &lt;- hpo$ancestors
# create HPO named list of genes
info &lt;- unstack(hpo_gene, geneID~HPO_ID)
hpByGene &lt;- unstack(hpo_gene, HPO_ID~geneID)

# information content
ic &lt;- descendants_IC(hpo)


## Compute semantic similarity between HP of interest and all HP terms
## This step is time consumming and can be parallelized.
SSscores &lt;- data.frame()
for (patient in unique(sa$SAMPLE_ID)){
  hpOfInterest &lt;- pat_hpo[SAMPLE_ID == patient]$HPO_ID
  
  hpGeneResnik &lt;- compareHPSets(
    hpSet1=names(ic), hpSet2=hpOfInterest,
    IC=ic,
    ancestors=hp_ancestors,
    method=&quot;Resnik&quot;,
    #BPPARAM= MulticoreParam(40)
    BPPARAM= MulticoreParam(snakemake@threads)
  )
  
  ## Group the results by gene
  hpMatByGene &lt;- lapply(
    hpByGene,
    function(x){
      hpGeneResnik[x, , drop=FALSE]
    }
  )
  
  ## Compute the corresponding scores
  Semantic_sim &lt;- unlist(lapply(
    hpMatByGene,
    hpSetCompSummary,
    method=&quot;bma&quot;, direction=&quot;symSim&quot;
  ))
  
  res &lt;- as.data.frame(Semantic_sim)
  res$geneID &lt;- rownames(res) 
  res$SAMPLE_ID &lt;- rep(  patient, nrow(res))
  res &lt;- as.data.table(res[, c(&quot;SAMPLE_ID&quot;, &quot;geneID&quot;, &quot;Semantic_sim&quot; )])
  res &lt;- res[order(Semantic_sim, decreasing = T)]
  res$Rank_SSs &lt;- seq(1, nrow(res))
  SSscores &lt;- rbind(SSscores, res)
}

hist(SSscores$Semantic_sim, xlab = &quot;Semantic similariy score&quot;,   
      main = &quot;Histogram of Semantic similariy scores&quot;, col = &quot;gray80&quot;, breaks = 30)</code></pre>
<p><img src="Scripts_MultiOmics_HPO_phenotype_Semantic_similarity_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<pre class="r"><code>#write_tsv(SSscores,  &#39;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/HPO/Patient_Gene_semantic_similariy.tsv&#39;)
write_tsv(SSscores,  snakemake@output$semantic_similariy)</code></pre>

<div id="rmd-source-code">IyctLS0KIycgdGl0bGU6IEdlbmUgc2VtYW50aWMgc2ltaWxhcml0eQojJyBhdXRob3I6IHNtaXJub3ZkCiMnIHdiOgojJyAgaW5wdXQ6CiMnICAtIGNvbmZpZzogJ3NyYy9jb25maWcuUicKIycgIC0gc2E6ICdgc20gY29uZmlnWyJBTk5PVEFUSU9OIl1gJwojJyAgLSBwYXRpZW50X2hwbzogJ2BzbSBjb25maWdbIlJBV19IUE8iXWAnCiMnICBvdXRwdXQ6CiMnICAtIHNlbWFudGljX3NpbWlsYXJpeTogJ2BzbSBjb25maWdbIlBST0NfREFUQSJdICsgIi9IUE8vUGF0aWVudF9HZW5lX3NlbWFudGljX3NpbWlsYXJpeS50c3YiYCcKIycgIHRocmVhZHM6IDQwCiMnIG91dHB1dDoKIycgICBodG1sX2RvY3VtZW50OgojJyAgICBjb2RlX2ZvbGRpbmc6IGhpZGUKIycgICAgY29kZV9kb3dubG9hZDogVFJVRQojJy0tLQoKCnNvdXJjZShzbmFrZW1ha2VAaW5wdXQkY29uZmlnKQojc291cmNlKCJzcmMvY29uZmlnLlIiKQoKCiMgUkVBRCBBTk5PVEFUSU9OCiMgc2EgPC0gZnJlYWQoJy9zL3Byb2plY3QvbWl0b011bHRpT21pY3MvbXVsdGlPTUlDc19pbnRlZ3JhdGlvbi9yYXdfZGF0YS9wcm90ZW9taWNzX2Fubm90YXRpb24udHN2JykKc2EgPC0gZnJlYWQoc25ha2VtYWtlQGlucHV0JHNhKQpzYSA8LSBzYVtVU0VfRk9SX1BST1RFT01JQ1NfUEFQRVIgPT0gVF0KCgojIGxvYWQgSFBPIG9udG9sb2d5CiMgZGF0YShocG8pIApocG8gPC0gZ2V0X29udG9sb2d5KCJodHRwOi8vcHVybC5vYm9saWJyYXJ5Lm9yZy9vYm8vaHAub2JvIiwgZXh0cmFjdF90YWdzPSJldmVyeXRoaW5nIikKCgojIExvYWQgcGF0aWVudCdzIEhQTyAKcGF0X2hwbyA8LSBmcmVhZChzbmFrZW1ha2VAaW5wdXQkcGF0aWVudF9ocG8pCiMgcGF0X2hwbyA8LSBmcmVhZCgnL3MvcHJvamVjdC9taXRvTXVsdGlPbWljcy9tdWx0aU9NSUNzX2ludGVncmF0aW9uL3Jhd19kYXRhL1BhdGllbnRfSFBPX3BoZW5vdHlwZXMudHN2JykKCgpwYXRfaHBvIDwtIHBhdF9ocG9bLCBjKCJTQU1QTEVfSUQiLCAiSFBPX0lEIildCiMgcmVubW92ZSBkdXBsaWNhdGVzIGZyb20gSFBPcyBwZXIgcGF0aWVudCAKcGF0X2hwbyA8LXBhdF9ocG9bIWR1cGxpY2F0ZWQocGF0X2hwbyApLF0KcGF0X2hwbyA8LXBhdF9ocG9bSFBPX0lEICVpbiUgIHVuaXF1ZShocG8kaWQpXQoKCmhwb19nZW5lIDwtIGZyZWFkKCJodHRwOi8vY29tcGJpby5jaGFyaXRlLmRlL2plbmtpbnMvam9iL2hwby5hbm5vdGF0aW9ucy9sYXN0U3VjY2Vzc2Z1bEJ1aWxkL2FydGlmYWN0L3V0aWwvYW5ub3RhdGlvbi9waGVub3R5cGVfdG9fZ2VuZXMudHh0IikKY29sbmFtZXMoaHBvX2dlbmUpIDwtIGMoIkhQT19JRCIsICJIUE9fbmFtZSIsICJlbnRyZXpJRCIsICJnZW5lSUQiLCAiQWRkaXRpb25hbF9JbmZvIiwgICJzb3VyY2UiLCAiZGlzZWFzZS1JRCIpCmhwb19nZW5lIDwtIGhwb19nZW5lWywgYygiZ2VuZUlEIiwgIkhQT19JRCIpXQpocG9fZ2VuZSA8LSBocG9fZ2VuZVshZHVwbGljYXRlZChocG9fZ2VuZSldCmhwb19nZW5lIDwtaHBvX2dlbmVbSFBPX0lEICVpbiUgIHVuaXF1ZShocG8kaWQpXQoKCiMgRml4IGdlbmUgbmFtZXMKaHBvX2dlbmVbZ2VuZUlEID09ICJZQVJTMSIsIGdlbmVJRCA6PSAiWUFSUyIgXQpocG9fZ2VuZVtnZW5lSUQgPT0gIk1JQ09TMTMiLCBnZW5lSUQgOj0gIkMxOU9SRjcwIiBdCmhwb19nZW5lW2dlbmVJRCA9PSAiQURQUlMiLCBnZW5lSUQgOj0gIkFEUFJITDIiIF0KCgojIEFkZCBrbm93biBnZW5vdHlwZSAtIHBoZW5vdHlwZSBhc3NvY2lhdGlvbnMgCmhwb19nZW5lIDwtIHJiaW5kKGhwb19nZW5lLCBkYXRhLnRhYmxlKCBnZW5lSUQgPSAiRE5BSkMzMCIsIEhQT19JRCA9IGMoIkhQOjAwMDIwNzYiLCAiSFA6MDAwMDY0OCIsICJIUDowMDAwNTA1IiwgIkhQOjAwMTE0NjIiLCAiSFA6MDAwMTExMiIsICJIUDowMDAzNTgxIiwgIkhQOjAwMDEyNjMiLCAiSFA6MDAwMTMzMiIsICJIUDowMDAxMjUxIiwgIkhQOjAwMDIxMzQiLCAiSFA6MDAxMTkyMyIsICJIUDowMDExNDYzIiAgKSAgKSApCmhwb19nZW5lIDwtIHJiaW5kKGhwb19nZW5lLCBkYXRhLnRhYmxlKCBnZW5lSUQgPSAiTVJQUzI1IiwgSFBPX0lEID0gYygiSFA6MDAwMzU4MSIsICJIUDowMDExNDQ1IiwgIkhQOjAwMDEyNzQiLCAiSFA6MDAwMjQyMSIsICJIUDowMDAxMjY2IiwgIkhQOjAwMDI1MDkiLCAiSFA6MDAwMTM0OCIsICJIUDowMDAxMjc0IiwgIkhQOjAwMDczMzMiLCAiSFA6MDAwMTMyNCIsIAogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIkhQOjAwMDEzODUiLCAiSFA6MDAwMDg0NiIsICJIUDowMDA0MzIyIiwgIkhQOjAwMDAyNTIiLCAiSFA6MDAxMTkyNCIsICJIUDowMDA4MzQ3IiwgIkhQOjAwMDEyNzAiLCAiSFA6MDAwMzY3NiIsICJIUDowMDEyMzc4IiwgIkhQOjAwMDIwMTUiKSAgKSApIApocG9fZ2VuZSA8LSByYmluZChocG9fZ2VuZSwgZGF0YS50YWJsZSggZ2VuZUlEID0gIlRYTklQIiwgSFBPX0lEID0gYygiSFA6MDAwNDkwMiIsICJIUDowMDAxOTQzIiwgIkhQOjAwMDMzNTUiLCAiSFA6MDAwNjU2OCIsICJIUDowMDAzNjU4IiwgIkhQOjAwMDQzMjIiLCAiSFA6MDAwMTUwOCIsIkhQOjAwMDEyNTIiLCAiSFA6MDAxMTk2OCIpICApICkKaHBvX2dlbmUgPC0gcmJpbmQoaHBvX2dlbmUsIGRhdGEudGFibGUoIGdlbmVJRCA9ICJZQVJTIiwgSFBPX0lEID0gYygiSFA6MDAwMTUxMCIsICJIUDowMDAxMjYzIiwgIkhQOjAwMDczMDUiLCAiSFA6MDAwMDQwNyIsICJIUDowMDAwNjM5IiwgIkhQOjAwMDI2MTEiLCAiSFA6MDAwMTczOCIsICJIUDowMDAxOTQzIiwgIkhQOjAwMDE5MDMiLCAiSFA6MDAwMDA5MyIsICJIUDowMTAwODA2IiwgIkhQOjAwMDY1MjgiKSAgKSApCgpocG9fZ2VuZSA8LSBocG9fZ2VuZVshZHVwbGljYXRlZChocG9fZ2VuZSldCnNhIDwtIHNhW1NBTVBMRV9JRCAlaW4lIHVuaXF1ZShwYXRfaHBvJFNBTVBMRV9JRCkgXQoKIyBleHRyYWN0IGFuY2VzdG9ycwpocF9hbmNlc3RvcnMgPC0gaHBvJGFuY2VzdG9ycwojIGNyZWF0ZSBIUE8gbmFtZWQgbGlzdCBvZiBnZW5lcwppbmZvIDwtIHVuc3RhY2soaHBvX2dlbmUsIGdlbmVJRH5IUE9fSUQpCmhwQnlHZW5lIDwtIHVuc3RhY2soaHBvX2dlbmUsIEhQT19JRH5nZW5lSUQpCgojIGluZm9ybWF0aW9uIGNvbnRlbnQKaWMgPC0gZGVzY2VuZGFudHNfSUMoaHBvKQoKCiMjIENvbXB1dGUgc2VtYW50aWMgc2ltaWxhcml0eSBiZXR3ZWVuIEhQIG9mIGludGVyZXN0IGFuZCBhbGwgSFAgdGVybXMKIyMgVGhpcyBzdGVwIGlzIHRpbWUgY29uc3VtbWluZyBhbmQgY2FuIGJlIHBhcmFsbGVsaXplZC4KU1NzY29yZXMgPC0gZGF0YS5mcmFtZSgpCmZvciAocGF0aWVudCBpbiB1bmlxdWUoc2EkU0FNUExFX0lEKSl7CiAgaHBPZkludGVyZXN0IDwtIHBhdF9ocG9bU0FNUExFX0lEID09IHBhdGllbnRdJEhQT19JRAogIAogIGhwR2VuZVJlc25payA8LSBjb21wYXJlSFBTZXRzKAogICAgaHBTZXQxPW5hbWVzKGljKSwgaHBTZXQyPWhwT2ZJbnRlcmVzdCwKICAgIElDPWljLAogICAgYW5jZXN0b3JzPWhwX2FuY2VzdG9ycywKICAgIG1ldGhvZD0iUmVzbmlrIiwKICAgICNCUFBBUkFNPSBNdWx0aWNvcmVQYXJhbSg0MCkKICAgIEJQUEFSQU09IE11bHRpY29yZVBhcmFtKHNuYWtlbWFrZUB0aHJlYWRzKQogICkKICAKICAjIyBHcm91cCB0aGUgcmVzdWx0cyBieSBnZW5lCiAgaHBNYXRCeUdlbmUgPC0gbGFwcGx5KAogICAgaHBCeUdlbmUsCiAgICBmdW5jdGlvbih4KXsKICAgICAgaHBHZW5lUmVzbmlrW3gsICwgZHJvcD1GQUxTRV0KICAgIH0KICApCiAgCiAgIyMgQ29tcHV0ZSB0aGUgY29ycmVzcG9uZGluZyBzY29yZXMKICBTZW1hbnRpY19zaW0gPC0gdW5saXN0KGxhcHBseSgKICAgIGhwTWF0QnlHZW5lLAogICAgaHBTZXRDb21wU3VtbWFyeSwKICAgIG1ldGhvZD0iYm1hIiwgZGlyZWN0aW9uPSJzeW1TaW0iCiAgKSkKICAKICByZXMgPC0gYXMuZGF0YS5mcmFtZShTZW1hbnRpY19zaW0pCiAgcmVzJGdlbmVJRCA8LSByb3duYW1lcyhyZXMpIAogIHJlcyRTQU1QTEVfSUQgPC0gcmVwKCAgcGF0aWVudCwgbnJvdyhyZXMpKQogIHJlcyA8LSBhcy5kYXRhLnRhYmxlKHJlc1ssIGMoIlNBTVBMRV9JRCIsICJnZW5lSUQiLCAiU2VtYW50aWNfc2ltIiApXSkKICByZXMgPC0gcmVzW29yZGVyKFNlbWFudGljX3NpbSwgZGVjcmVhc2luZyA9IFQpXQogIHJlcyRSYW5rX1NTcyA8LSBzZXEoMSwgbnJvdyhyZXMpKQogIFNTc2NvcmVzIDwtIHJiaW5kKFNTc2NvcmVzLCByZXMpCn0KCmhpc3QoU1NzY29yZXMkU2VtYW50aWNfc2ltLCB4bGFiID0gIlNlbWFudGljIHNpbWlsYXJpeSBzY29yZSIsICAgCiAgICAgIG1haW4gPSAiSGlzdG9ncmFtIG9mIFNlbWFudGljIHNpbWlsYXJpeSBzY29yZXMiLCBjb2wgPSAiZ3JheTgwIiwgYnJlYWtzID0gMzApCgojd3JpdGVfdHN2KFNTc2NvcmVzLCAgJy9zL3Byb2plY3QvbWl0b011bHRpT21pY3MvbXVsdGlPTUlDc19pbnRlZ3JhdGlvbi9wcm9jZXNzZWRfZGF0YS9IUE8vUGF0aWVudF9HZW5lX3NlbWFudGljX3NpbWlsYXJpeS50c3YnKQp3cml0ZV90c3YoU1NzY29yZXMsICBzbmFrZW1ha2VAb3V0cHV0JHNlbWFudGljX3NpbWlsYXJpeSkKCg==</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("Semantic_similarity.R");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
